updater/fetchers/alpine/alpine.go

```diff
@@ -0,0 +1,293 @@
+)
+
+const (
+	// When available, this should be updated to use HTTPS.
+	secdbGitURL  = "http://git.alpinelinux.org/cgit/alpine-secdb"
+	updaterFlag  = "alpine-secdbUpdater"
+	nvdURLPrefix = "https://cve.mitre.org/cgi-bin/cvename.cgi?name="
+)
+
+var (
+	// ErrFilesystem is returned when a fetcher fails to interact with the local filesystem.
+	ErrFilesystem = errors.New("updater/fetchers: something went wrong when interacting with the fs")
+
+	// ErrGitFailure is returned when a fetcher fails to interact with git.
+	ErrGitFailure = errors.New("updater/fetchers: something went wrong when interacting with git")
+
+	log = capnslog.NewPackageLogger("github.com/coreos/clair", "updater/fetchers/alpine")
+)
+
+func init() {
+	updater.RegisterFetcher("alpine", &fetcher{})
+}
+
+type fetcher struct {
+	repositoryLocalPath string
+}
+
+func (f *fetcher) FetchUpdate(db database.Datastore) (resp updater.FetcherResponse, err error) {
+	log.Info("fetching Alpine vulnerabilities")
+
+	// Pull the master branch.
+	var commit string
+	commit, err = f.pullRepository()
+	if err != nil {
+		return
+	}
+
+	// Ask the database for the latest commit we successfully applied.
+	var dbCommit string
+	dbCommit, err = db.GetKeyValue(updaterFlag)
+	if err != nil {
+		return
+	}
+
+	// Set the updaterFlag to equal the commit processed.
+	resp.FlagName = updaterFlag
+	resp.FlagValue = commit
+
+	// Short-circuit if there have been no updates.
+	if commit == dbCommit {
+		log.Debug("no alpine update")
+		return
+	}
+
+	var namespaces []string
+	namespaces, err = detectNamespaces(f.repositoryLocalPath)
+	// Append any changed vulnerabilities to the response.
+	for _, namespace := range namespaces {
+		var vulns []database.Vulnerability
+		vulns, err = parseVulnsFromNamespace(f.repositoryLocalPath, namespace)
+		if err != nil {
+			return
+		}
+		resp.Vulnerabilities = append(resp.Vulnerabilities, vulns...)
+	}
+
+	return
+}
+
+func detectNamespaces(path string) ([]string, error) {
+	// Open the root directory.
+	dir, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	defer dir.Close()
+
+	// Get a list of the namspaces from the directory names.
+	names, err := dir.Readdirnames(0)
+	if err != nil {
+		return nil, err
+	}
+
+	var namespaces []string
+	for _, name := range names {
+		// Filter out hidden directories like `.git`.
+		if strings.HasPrefix(name, ".") {
+			continue
+		}
+
+		namespaces = append(namespaces, name)
+	}
+
+	return namespaces, nil
+}
+
+type parserFunc func(io.Reader) ([]database.Vulnerability, error)
+
+var parsers = map[string]parserFunc{
+	"v3.3": parse33YAML,
+	"v3.4": parse34YAML,
+}
+
+func parseVulnsFromNamespace(repositoryPath, namespace string) (vulns []database.Vulnerability, err error) {
+	var file io.ReadCloser
+	file, err = os.Open(repositoryPath + "/" + namespace + "/main.yaml")
+	if err != nil {
+		return
+	}
+	defer file.Close()
+
+	parseFunc, exists := parsers[namespace]
+	if !exists {
+		return
```

Comment:
All these are much better. Please just add a Warning here as well as an updater Note for that so we can be alerted when we need to take action. See https://github.com/coreos/clair/blob/master/updater/fetchers/ubuntu/ubuntu.go#L139